src/Interface/HasFilter/InstanceTh.agda

```diff
@@ -0,0 +1,20 @@
+{-# OPTIONS --safe --no-import-sorts #-}
+open import Agda.Primitive renaming (Set to Type)
+open import Axiom.Set using (Theory)
+
+module Interface.HasFilter.InstanceTh {th : Theory} where
+
+open import Data.Product using (_×_; _,_)
+open import Data.List using (List) renaming (filter to filterˡ)
+open import Relation.Unary using (Decidable)
+open import Axiom.Set.Map th
+open import Interface.HasFilter
+open Theory th renaming (filter to filterˢ)
+private variable A B : Type
+
+instance
+  mapFilter : HasFilter (A × B) (Map A B)
+  mapFilter = record { predCheck = specProperty; filter = filterᵐ}
+
+  setFilter : HasFilter A (Set A)
+  setFilter = record { predCheck = specProperty; filter = filterˢ}
```

Comment:
I'm confused about when I need to have an explicit theory (`th`) in context.  I don't see how to create these filter instances for `Map` and `Set` unless I have a `th` around.  And, assuming this is correct, and I need to parameterize this module with a `th`, how do I get it to work with, e.g., the `Transaction` module which does not have an explicit `th`?

Reply:
It comes from `Ledger.Set.Theory`. So if you have a module that's quantified over a set model, you could do `open Interface.HasFilter.InstanceTh {List-Model} public`. Probably the best place to do this is in `Ledger.Prelude`, it already has some instance imports like that at the bottom.

Reply:
Okay, I'll give that a try.  Thanks.